walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -108,4 +109,83 @@
     Collections.sort(timeConflicts, TimeRange.ORDER_BY_START);
     return timeConflicts;
   }
+
+  /**
+   * Find the optimal time ranges for optional attendees.
+   *
+   * @param events
+   * @param request
+   * @return Returns the optimal time ranges for optional attendees. Can be empty.
+   */
+  private Collection<TimeRange> findOptimalOptionalAttendee(
+      Collection<Event> events, MeetingRequest request) {
+    // This works by checking each combination of optional attendees in increasing quantity.
+    // Starting with groups of one attendee, then two, etc. Until less than two combinations
+    // had non-empty ranges (see optimization notes below) or all combinations have been checked.
+    // 
+    // For large numbers of optional attendees, this can quickly become inefficient.
+    // There is room here for a dynamic programming approach following the logic that a combination
+    // of three elements could be more efficiently computed by using a previously-computed
+    // combination of two elements. 
+
+    ArrayList<String> optionalAttendees = 
+        new ArrayList<String>(request.getOptionalAttendees());
+    Collection<String> attendees = request.getAttendees();
+
+    int successfulCombinations = 0;
```

Comment:
Oh wow that’s awkward. Thanks for catching that. Unlucky coincidence that it broke in a way that made the optimization less conservative, so it passed all of the testing. 

Also, thanks for that resource. Gave it a read and it’s very helpful. Definitely bookmarking that! 

However as for removing it, I wonder if it falls into partially functional and partially optimization? I think it’d be sensible to have a break case if there are no successful combinations (as that logically means there cannot be higher quantity combinations that are successful) so the only difference is checking for 2 instead of 1. Is there a way that I could keep this while avoiding the perils of unreadability?